x-pack/plugin/core/src/main/resources/beats-agents-template.json

```diff
@@ -0,0 +1,54 @@
+{
+  "index_patterns": [
+    ".management-beats-agents"
+  ],
+  "settings": {
+    "index": {
+      "number_of_shards": 1,
+      "auto_expand_replicas": "0-1",
+      "codec": "best_compression"
+    }
+  },
+  "mappings": {
+    "_doc": {
+      "_meta": {
+        "beats-version": "${beats.template.version}"
+      },
+      "dynamic": "strict",
+      "properties": {
+        "beat": {
+          "properties": {
+            "id": {
+              "type": "keyword"
+            },
+            "enrollment_token": {
+              "type": "keyword"
+            },
+            "access_token": {
+              "type": "keyword"
+            },
+            "verified_on": {
+              "type": "date"
+            },
+            "type": {
+              "type": "keyword"
+            },
+            "ephemeral_id": {
+              "type": "keyword"
+            },
+            "local_configuration_yml": {
+              "type": "text"
+            },
+            "central_configuration_yml": {
```

Comment:
We don't have to but I think it makes sense to do it from a performance perspective. Basically we can compute this value whenever anytime changes are made in the UI, which should be relatively less frequent compared to the how often a Beat agent might request this value. So I think pre-computing it and storing it means we don't have to compute it on the fly each time a Beat requests it.

Are you thinking there is some downside of storing this? We can turn indexing off on this field if storage space is a concern (it will still be available via `_source` then).

Reply:
Got it, makes sense

We definitely need some kind of caching for confs, so we either do it in memory (one config per tag combination, no need to have one per beat) or store them here. I have one concern with this approach: Changing a tag requires updating all beats on that tag, that means going one by one to get their tag list, combine blocks and update the full `central_configuration_yml`.

Reply:
>  I have one concern with this approach: Changing a tag requires updating all beats on that tag, that means going one by one to get their tag list, combine blocks and update the full `central_configuration_yml`.

This is true. However, I think:
1. This should be an infrequent operation compared to how often a Beat will request its (merged) configuration,
2. We can get all the Beats for a tag from ES in a single `_search` request, and
3. We can update all the Beats' `central_configuration_yml` (after it is recomputed) in ES with a single `_bulk` request.

Let's see how all of this shakes out after an initial implement. I plan to use something like siege or httperf to simulate 10000s of Beats requesting configs while changing a tag in the midst of that scenario. We can optimize the long poles in the performance tent as we go.